modules/utxo-lib/src/bitgo/UtxoPsbt.ts

```diff
@@ -68,26 +68,27 @@ export class UtxoPsbt<Tx extends UtxoTransaction<bigint>> extends Psbt {
     return psbt;
   }
 
-  static async fromTransactionComplete(
-    transaction: UtxoTransaction<bigint>,
-    prevOutputs: TxOutput<bigint>[],
-    fetchTransactions: (txids: string[]) => Promise<Record<string, Buffer>>
-  ): Promise<UtxoPsbt<UtxoTransaction<bigint>>> {
-    const psbt = this.fromTransaction(transaction, prevOutputs);
-
-    const txidToIndex: Record<string, number> = {};
-    psbt.data.inputs.forEach((input, index) => {
-      if (!input.tapLeafScript && !input.witnessScript) {
-        txidToIndex[reverseBuffer(transaction.ins[index].hash).toString('hex')] = index;
+  async addNonWitnessUtxos(fetchTransactions: (txids: string[]) => Promise<Record<string, Buffer>>): Promise<this> {
```

Comment:
I am not a fan of `async` here. It has the nasty habit to propagate up the call chain.

I think we can realize this with a sync signature `addNonWitnessUtxos(txBuffers: Map<string, Buffer>)` instead.

It would be more consistent with the way we handle `prevOutputs: TxOutput[]`.

To help the caller figure out which transactions to fetch we can add a method `getNonWitnessInputs(): { index: number, input: TxInput }[]`

Reply:
reason to use `Map` instead of `Record`: https://stackoverflow.com/a/18541990

Reply:
also a docsting explaining the purpose would be nice